.github/workflows/main.yaml

```diff
@@ -161,6 +161,9 @@ jobs:
         with:
           targets: ${{ needs.initialize.outputs.targets }}
           test_artifacts_key: ${{ env.TEST_ARTIFACTS_KEY }}
+      - name: Upload On Device Test Artifacts
+        if: ${{ (startsWith(matrix.platform, 'android') && matrix.config == 'devel') }}
```

Comment:
Instead of using `startsWith(matrix.platform, 'android'` you should use `needs.initialize.outputs.test_on_device == 'true'`

Reply:
done